File: go/002/main.go
/*
 * ProjectEuler 002
 * Go / main.go
 * copyright (c) 2015 Susisu
 *
 * Go 1.4.1
 * `go build -o bin/main main.go`
 */


package main

import "fmt"

func main() {
    sum := 0
    a := 0
    b := 1
    for a <= 4000000 {
        t := a
        a  = b
        b += t
        if a % 2 == 0 {
            sum += a
        }
    }
    fmt.Printf("%d\n", sum)
}

File: scala/002/Main.scala
/*
 * ProjectEuler 002
 * Scala / Main.scala
 * copyright (c) 2015 Susisu
 *
 * Scala 2.11.4
 * `scalac Main.scala -d bin`
 */

object Main {
    def main(args: Array[String]) {
        var sum = 0;
        var a = 0;
        var b = 1;
        while (a <= 4000000) {
            var t = a;
            a     = b;
            b    += t;
            if (a % 2 == 0) {
                sum += a;
            }
        }
        println(sum);
    }
}
